### DISCUSSION TITLE
Button setting for new service

### DISCUSSION THREAD

**Discussion by oulom** [InseeFrLab/onyxia#855] [Q&A]
Hello, 

Just starting out. I'm trying to create a new service. 

When the service is launched, there's a dialog box that displays the contents of the NOTES.txt file, but I can't see where I should set the buttons (for example, click to copy the password, etc.).

Can you help me?

**Comment by ihiverlet** [comment]
Hello,
There are two mechanisms. When the NOTES.txt contains the default service password, it is detected and the button 'click to copy the password' is displayed. 
Otherwise, the NOTES.txt must contain `password:  [password] `  so that onyxia can detect it using a regex and display the button.
You can find the regex associated with this behaviour [here](https://github.com/InseeFrLab/onyxia/blob/760b747e21979208d9f4397d09d1e53e24b40b69/web/src/ui/pages/myServices/MyServicesCards/MyServicesCard/ReadmeDialog/ReadmeDialog.tsx#L138)

**Comment by oulom** [comment]
Thank you for your answer. So what I understand is : we have some sort of
default button depending of what is inside the NOTES.txt (right ?)
So for the 'URL Link's it is the same ?

Le ven. 13 sept. 2024 à 13:14, ihiverlet ***@***.***> a
écrit :

**Reply by ihiverlet** [reply]
Yes, it takes the first url given the alphabetic order

**Reply by oulom** [reply]
That's right, it's alphabetical order. How is the url for the note determined? I'm asking because after the domain, the software adds “null” to the url.

**Reply by ihiverlet** [reply]
It takes the url when looking up the ingresses

**Comment by garronej** [comment]
Here is the full arlgorithm: https://github.com/InseeFrLab/onyxia/issues/838#issuecomment-2256135263
